addons/hr_evaluation/models/hr_evaluation.py

```diff
@@ -0,0 +1,402 @@
+
+    def _subscribe_users(self):
+        user_ids = [emp.user_id.id for emp in self.appraisal_manager_ids if emp.user_id]
+        if self.employee_id.user_id:
+            user_ids.append(self.employee_id.user_id.id)
+        if self.employee_id.department_id.manager_id.user_id:
+            user_ids.append(self.employee_id.department_id.manager_id.user_id.id)
+        if self.employee_id.parent_id.user_id:
+            user_ids.append(self.employee_id.parent_id.user_id.id)
+        return self.message_subscribe_users(user_ids=user_ids)
+
+    @api.model
+    def create(self, vals):
+        res = super(hr_evaluation, self.with_context(mail_create_nolog=True)).create(vals)
+        if res.appraisal_manager_ids:
+            res._subscribe_users()
+            res.message_post(body=_("Employee Appraisal created"), subtype="mail.mt_comment", type="notification")
+        return res
+
+    @api.multi
+    def write(self, vals):
+        emp_obj = self.env['hr.employee']
+        for evl_rec in self:
+            # avoid recursive process
+            if vals.get('state') == 'pending' and not evl_rec._context.get('send_mail_status'):
+                evl_rec.button_send_appraisal()
+            if vals.get('interview_deadline') and self.state == 'pending' and not vals.get('meeting_id'):
+                # creating employee meeting and interview date
+                evl_rec.create_update_meeting(vals)
+            if vals.get('appraisal_manager_ids'):
+                # add followers
+                user_ids = [employee.user_id.id for employee in emp_obj.browse(vals['appraisal_manager_ids'][0][2]) if employee.user_id]
+                evl_rec.message_subscribe_users(user_ids)
+        return super(hr_evaluation, self).write(vals)
+
+    @api.one
+    def create_token(self, email, survey, partner_id):
+        """ Create response with token """
+        token = uuid.uuid4().__str__()
+        self.env['survey.user_input'].create({
+            'survey_id': survey.id,
+            'deadline': self.date_close,
+            'date_create': datetime.datetime.now().strftime(DF),
+            'type': 'link',
+            'state': 'new',
+            'token': token,
+            'evaluation_id': self.id,
+            'partner_id': partner_id.id if partner_id else None,
+            'email': email})
+        return token
+
+    @api.multi
+    def send_appraisal(self, appraisal_receiver):
+        mail_obj = self.env['mail.thread']
+        mail_template_obj = self.env['mail.template']
+        for record in appraisal_receiver:
+            for emp in record['employee_ids']:
+                email = tools.email_split(emp.work_email) and tools.email_split(emp.work_email)[0] or False
+                #'_find_partner_from_emails' is decorated with @api.one so here used [0][0]
+                if email:
+                    partner_id = mail_obj._find_partner_from_emails(email) or emp.user_id.partner_id or None
```

Comment:
- are you sure about the return format of `_find_partner_from_emails`? list of ids ?
- remove `or None` clause
- if partner_id is empty, raise an error asking to configure the employee